docs/source/reference/pragmas.rst

```diff
@@ -124,3 +226,247 @@ You can also hide fixity declarations if you need to redefine your own.
    %hide Prelude.Ops.infixl.(+)
 
    infixr 5 +
+
+
+``%unhide``
+--------------------
+
+The ``%unhide`` pragma unhides a definition that was previously hidden with ``%hide``.
+
+
+``%unsafe``
+--------------------
+
+Mark a function like ``believe_me`` as being unsafe. The function will be semantically
+highlighted in a different color to draw the user's attention to its use.
+
+
+``%spec``
+--------------------
+
+Specialise a function according to a list of arguments.
+
+.. code-block:: idris
+
+   %spec a
+   identity : List a -> List a
+   identity [] = []
+   identity (x :: xs) = x :: identity xs
+
+
+``%foreign``
+--------------------
+
+Declare a foreign function.  It is followed by an indented block of expressions
+that evaluate to strings. See :ref:`ffi-overview` for more details.
+
+``%export``
+--------------------
+
+Export an Idris function to the underlying host language. The the name for each backend is
+given in an indented block of string expressions, similar to ``%foreign``.  Currently this
+pragma is only supported by the javascript backend.
+
+.. code-block:: idris
+
+   %export "javascript:addNat"
+   addNat : Nat -> Nat -> Nat
+   addNat a b = a + b
+
+
+``%nomangle``
+--------------------
+
+This pragma is deprecated.  Instead use ``%export`` to expose functions to the backend.
+
+
+``%hint``
+--------------------
+
+Mark a function to be used for ``auto`` search (see :ref:`auto-implicits` for more).
```

Comment:
Maybe, it's important to add that every non-named implementation receives this pragma automatically. Also, it may be added that every data constructor effectively receives this unless `[noHints]` is present in the declaration of the data type.